observability/metrics/src/main/scala/com/daml/metrics/ExecutorServiceMetrics.scala

```diff
@@ -162,4 +192,55 @@ object ExecutorServiceMetrics {
 
   }
 
+  /*
+   * Handles the removal of registered gauges for the executor service.
+   * As all the gauges registered are async the removal could've been also done when
+   * trying to read the value of a gauge, but this would not prevent warnings being logged when an executor with
+   * the same name is registered before the gauges for a previous one was removed.
+   * */
+  class MetricResourceCleaningExecutorService(
+      val delegate: ExecutorService,
+      resourceCleaning: AutoCloseable,
```

Comment:
Naming suggestion (i.e., to better reflect intent):
```suggestion
  class   class ExecutorServiceWithCleanup(
      val delegate: ExecutorService,
      resourceCleanup: AutoCloseable,
```